Phase 2

PranavShahSI opened the pull request:
- Add Extensions
- Add Dialogs
- Add Preference
- Add Interceptors
- Add Base Classes
- Add Binding Adapters
- Update Dependency
- Update Project Version
- Add Constant
- Add Models
- Add Exception
- Add Modules (Manager, Network)
- Add Interface
- Add Print Log
- Add provider Path
- Add Resource Provider
- Add Single Live Event

PranavShahSI committed:
- add activity extension
- add interceptors(auth, network)
- add base classes (fragment, activity, view model, adapter)
- add binding adapters
- update dependency
- update project version
- add common preference
- add constant file
- add date extension
- add message dialog
- add progress dialog
- add image picker dialog
- add error response model
- add exceptions
- add extensions
- add fragment extensions
- add manager module
- add network module
- add interfaces (onDialogCall, onDialogDetailCall, OnDismissedCall, OnItemSelected, OnMultipleItemSelected)
- add permission extension
- add print log
- add provide path
- add resource provider
- add response model
- add single live event
- add string extension
- add strings
- add style
- add view extension

PranavShahSI committed:
- add get/set model & array from preference
- add function for dialog for get values
- add file picker permission interface
- add confirmation dialog

PranavShahSI committed:
version upgrade

PranavShahSI committed:
minor change

PranavShahSI committed:
minor change

vivekpshaligram commented on sit-common/src/main/java/com/sit/common/api/interceptor/NetworkConnectionInterceptor.kt:
```diff
@@ -0,0 +1,45 @@
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        if (!isInternetAvailable())
+            throw NoInternetException(Constant.NO_INTERNET)
```
Please use strings from the strings.xml file so we can adjust the messages according to the app's needs.

vivekpshaligram commented on sit-common/src/main/java/com/sit/common/api/interceptor/AuthInterceptor.kt:
```diff
@@ -0,0 +1,21 @@
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val newRequest: Request.Builder = chain.request().newBuilder()
+        if (preferenceManager.authToken.isNotEmpty()) {
+            newRequest.addHeader(Constant.AUTHORIZATION, preferenceManager.authToken)
```
Please use strings from the strings.xml file so we can adjust the messages according to the app's needs.

PranavShahSI committed:
- remove some constant and add in strings.xml
- manage dynamically snack bar gravity
- create abstract function for per page value update
- remove unnecessary code
- create error message binding adapter
- update common preference name
- add some date extensions
- add event channel
- update extensions
- add some view extensions

PranavShahSI committed:
update readme file

PranavShahSI committed:
- update read me file
- change normal share preference to encryptedSharedPreference
